nistats/second_level_model.py

```diff
@@ -227,69 +216,41 @@ def fit(self, second_level_input, first_level_conditions=None,
         if design_matrix is not None:
             if not isinstance(design_matrix, pd.DataFrame):
                 raise ValueError('design matrix must be a pandas DataFrame')
+            if 'contrast' not in design_matrix.columns:
+                raise ValueError('design matrix must contain column "contrast"')
 
-        # Build the design matrix X and list of imgs Y for GLM fit
+        self.second_level_input_ = second_level_input
+        self.confounds_ = confounds
+
+        # Report progress
+        t0 = time.time()
+        if self.verbose > 0:
+            sys.stderr.write("Fitting second level model. "
+                             "Take a deep breath\r")
+
+        # Select sample map for masker fit and get maps_table for design
         if isinstance(second_level_input, pd.DataFrame):
+            sample_map = second_level_input['effects_map_path'][0]
             maps_table = second_level_input
-            # Get only first level conditions if provided
-            if first_level_conditions is not None:
-                for name, condition in first_level_conditions:
-                    if condition not in maps_table['map_name'].tolist():
-                        raise ValueError('condition %s not present in'
-                                         ' second_level_input' % condition)
-                condition_list = [cond[0] for cond in first_level_conditions]
-                in_cond = maps_table.apply(
-                    lambda x: x['map_name'] in condition_list, axis=1)
-                maps_table = maps_table[in_cond]
-            # Create design matrix if necessary
-            if design_matrix is None:
-                design_matrix = create_second_level_design(maps_table,
-                                                           confounds)
-            # get effect maps for fixed effects GLM
-            effects_maps = maps_table['effects_map_path'].tolist()
-
         elif isinstance(second_level_input[0], FirstLevelModel):
-            # Check models were fit
-            for model_idx, model in enumerate(second_level_input):
-                if model.labels_ is None:
-                    raise ValueError(
-                        'Model at idx %d has not been fit' % model_idx)
-            # Get the first level model maps
+            sample_model = second_level_input[0]
+            sample_condition = sample_model.design_matrices_[0].columns[0]
+            sample_map = sample_model.compute_contrast(
+                sample_condition, output_type='effect_size')
```

Comment:
I agree that 'effect size' is the reight default choice, but it has to be documented. At that point I'm not sure i would like to support other outpute (e.g. z maps etc.)